src/openforms/js/components/admin/form_design/registrations/zgw/fields/CatalogueSelect.js

```diff
@@ -0,0 +1,60 @@
+import {useFormikContext} from 'formik';
+import {FormattedMessage} from 'react-intl';
+import {useAsync} from 'react-use';
+
+import {
+  CatalogueSelect as GenericCatalogueSelect,
+  groupAndSortCatalogueOptions,
+} from 'components/admin/forms/zgw';
+import {get} from 'utils/fetch';
+
+// Data fetching
+
+const CATALOGUES_ENDPOINT = '/api/v2/registration/plugins/zgw-api/catalogues';
+
+const getCatalogues = async apiGroupID => {
+  const response = await get(CATALOGUES_ENDPOINT, {zgw_api_group: apiGroupID});
+  if (!response.ok) {
+    throw new Error('Loading available catalogues failed');
+  }
+  return groupAndSortCatalogueOptions(response.data);
+};
+
+// Components
+
+const CatalogueSelect = () => {
```

Comment:
Again confused..why do we have two separate components (one in objectsAPI and one in ZGW)?Perhaps it has to do with my question above concerning the serializers in the backend.

Reply:
both use the same generic `components/admin/forms/zgw/CatalogueSelect`, so they're both specializations of that one.

They are different because the registration plugins are different - objects API should not be aware of ZGW or vice versa. Additionally, the context of each select is different for each plugin, which can result in different form field labels, help texts or even catalogue selection options. In particular here, the catalogue select is disabled if no API group has been selected yet, and for Objects API the form field is `objectsApiGroup`, while for ZGW APIs the form field for the API group is `zgwApiGroup`. Additionally, the available options are loaded from their plugin-specific API endpoints.